lib/expression/v2/compat.ex

```diff
@@ -115,62 +114,56 @@ defmodule Expression.V2.Compat do
 
-    # v2_resp =
-    #   V2.eval(
-    #     expression,
-    #     V2.Context.new(patch_v1_context(context), callback_module)
-    #   )
-    #   |> hd
+    v2_resp =
+      V2.eval(
+        expression,
+        V2.Context.new(patch_v1_context(context), callback_module)
+      )
+      |> hd
 
-    # return_or_raise(expression, context, v1_resp, v2_resp)
-    unpack_returned_value(v1_resp)
+    return_or_log(expression, context, v1_resp, v2_resp)
   end
 
   def evaluate_block!(
         expression,
         context \\ %{},
-        callback_module \\ Callbacks.Standard
+        callback_module \\ V2.Callbacks.Standard
       )
 
   def evaluate_block!(expression, context, callback_module) do
-    v1_resp = Expression.evaluate_block(expression, context, callback_module)
-
-    # v2_resp =
-    #   case V2.eval_block(
-    #          expression,
-    #          V2.Context.new(patch_v1_context(context), callback_module)
-    #        ) do
-    #     {:error, error, reason} -> {:error, error <> " " <> reason}
-    #     value -> {:ok, value}
-    #   end
-
-    # cond do
-    #   # Hack for handling random returns from `rand_between()` callback function
-    #   # these will throw an error because they're designed to be different every time
-    #   String.contains?(expression, "rand_between") ->
-    #     return_or_raise(expression, context, v2_resp, v2_resp)
-
-    #   # Hack for handling `@if` expressions, in V2 these aren't evaluated.
-    #   # See the note for this in `eval_compat_test.exs`.
-    #   String.contains?(String.downcase(expression), ["@if", "@left"]) ->
-    #     return_or_raise(expression, context, v2_resp, v2_resp)
-
-    #   true ->
-    #     return_or_raise(expression, context, v1_resp, v2_resp)
-    # end
-    unpack_returned_value(v1_resp)
-  end
+    v1_resp = Expression.evaluate_block(expression, context, v1_module(callback_module))
 
-  def unpack_returned_value({:ok, val}), do: val
-  def unpack_returned_value({:error, reason}), do: reason
-  def unpack_returned_value(other), do: other
+    v2_resp =
+      case V2.eval_block(
+             expression,
+             V2.Context.new(patch_v1_context(context), callback_module)
+           ) do
+        {:error, error, reason} -> {:error, error <> " " <> reason}
+        value -> {:ok, value}
+      end
+
+    cond do
+      # Hack for handling random returns from `rand_between()` callback function
+      # these will throw an error because they're designed to be different every time
+      String.contains?(expression, "rand_between") ->
+        return_or_log(expression, context, v2_resp, v2_resp)
+
+      # Hack for handling `@if` expressions, in V2 these aren't evaluated.
+      # See the note for this in `eval_compat_test.exs`.
+      String.contains?(String.downcase(expression), ["@if", "@left"]) ->
+        return_or_log(expression, context, v2_resp, v2_resp)
+
+      true ->
+        return_or_log(expression, context, v1_resp, v2_resp)
+    end
```

Comment:
I think this can be simplified as we are not doing anything with the different conditions
```suggestion
    return_or_log(expression, context, v1_resp, v2_resp)
```

Reply:
the difference is we're defaulting to whatever v2 returns here rather than `v1_resp` or `v2_resp`. The difference is subtle but important.

Reply:
Ahh got it! missed that. Thanks @smn